src/main/java/com/minecolonies/coremod/event/TagWorkAroundEventHandler.java

```diff
@@ -104,19 +108,56 @@ public static void onDataPackSync(final OnDatapackSyncEvent event)
 
     public static class TagClientEventHandler
     {
+        private static int eventCounter = 0;
+        private static RecipeManager recipeManager;
+
+        /**
+         * The RecipesUpdatedEvent and TagsUpdatedEvent occur in unspecified order on the client side
+         * (or rather, a different order for initial connect vs. /reload) so we have to handle them
+         * happening either way around -- but we also don't want to trigger our recipe reload sequence
+         * until we have both of them.
+         *
+         * @param maybeRecipeManager null or a new RecipeManager to temporarily cache
+         */
+        private static void maybeLoadRecipes(@Nullable final RecipeManager maybeRecipeManager)
+        {
+            if (maybeRecipeManager != null)
+            {
+                recipeManager = maybeRecipeManager;
+            }
+
+            ++eventCounter;
```

Comment:
This needs to be reset on server leave.

Reply:
It's reset when it hits 2 (which should always happen unless they somehow disconnect between the two packets, in which case they presumably wouldn't get a leave event either?).  Or if something goes weird, that's why it checks for a non-null recipeManager too.